### PR TITLE
Fix all PEP-008 issues

### PR DESCRIPTION
**Pull request by nsoranzo**
and add a Travis job for flake8 to check future PRs.

### CODE REVIEW

File: cibuildwheel/linux.py [hunk 1]
```diff
@@ -33,7 +38,7 @@ def get_python_configurations(build_selector):
 def build(project_dir, output_dir, test_command, test_requires, test_extras, before_build, build_verbosity, build_selector, environment, manylinux_images):
     try:
         subprocess.check_call(['docker', '--version'])
-    except:
+    except Exception:
```
Comment (paulmelnikow) [inline cibuildwheel/linux.py]:
```suggestion
    except subprocess.CalledProcessError:
```

**Comment by nsoranzo** [comment]
Rebased to fix conflicts.

**Comment by joerick** [comment]
I would like to get something like this in cibuildwheel, but it would be good to choose the right time when we don't have lots of open PRs!

Did you generate this PR using a tool like autopep8, @nsoranzo ? that might be handy for existing PRs and to document. Also, I'd kinda be in favour of putting the flake8 checker on CircleCI, since that on tends to be quicker, and is listed separately in Github.

**Comment by nsoranzo** [comment]
> I would like to get something like this in cibuildwheel, but it would be good to choose the right time when we don't have lots of open PRs!

Up to you, let me know how you would like to proceed.

> Did you generate this PR using a tool like autopep8, @nsoranzo ? that might be handy for existing PRs and to document.

Not in this case. I did use autopep8 for a larger codebase, but only for some error codes.

> Also, I'd kinda be in favour of putting the flake8 checker on CircleCI, since that on tends to be quicker, and is listed separately in GitHub.

Done.

**Comment by nsoranzo** [comment]
Rebased to fix some conflicts.

**Comment by nsoranzo** [comment]
The Travis job failure is unrelated.

**Comment by nsoranzo** [comment]
Rebased.

**Comment by nsoranzo** [comment]
Rebased with some new fixes.

**Comment by YannickJadoul** [comment]
@joerick If you intend to merge this, we should do it soon enough, so we relieve @nsoranzo from the endless rebasing ;-)

**Comment by joerick** [comment]
Ah, I agree (sorry @nsoranzo), I'm just aware that we've got a lot of PRs active right now and it'll cause conflicts with ~likely all of them! I was planning to merge this once #156 and the related PRs #220, #194 go in, along with whatever else we can get in at the same time.

**Comment by YannickJadoul** [comment]
Build failures are linked to #270 & virtualenv.

@joerick Shouldn't we merge this soon? #156 and #220 are merged, and we can't expect @nsoranzo to keep on rebasing and fixing more PEP8 errors? Maybe it's time to push that burden to the developers of individual PR (which unfortunately includes myself, but OK) ?

**Comment by nsoranzo** [comment]
@YannickJadoul Thanks for the heads-up, I was struggling to find how I broke it :smile: 

Obviously happy if this gets merged!

**Comment by joerick** [comment]
Apologies, I'm busy with work this week and don't have time to look properly, but I think I'd agree with you @yannickjadoul so please feel free to merge things as you see fit!

**Comment by YannickJadoul** [comment]
If you rebase one last time (onto the merged #270), @nsoranzo, I'll finally get this in. (After which I probably need to start cleaning up all my outstanding PRs, but OK... :-D )

**Comment by nsoranzo** [comment]
@YannickJadoul Done!

**Comment by YannickJadoul** [comment]
Merged! Let's see how many PRs will be broken :-D

**Comment by YannickJadoul** [comment]
Thanks a lot the this PR and the countless rebases, btw!

**Comment by joerick** [comment]
+1 Thanks @nsoranzo !

**Pull request merged**